test/Roave/SecurityAdvisories/FriendsOfPhpSpec.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Roave.SecurityAdvisories.FriendsOfPhpSpec (main, spec) where
+
+import Data.Map (singleton)
+import Data.ByteString (ByteString)
+import Text.RawString.QQ
+
+import Test.Hspec
+
+import Roave.SecurityAdvisories.FriendsOfPhp
+
+
+-- `main` is here so that this module can be run from GHCi on its own.  It is
+-- not needed for automatic spec discovery.
+main :: IO ()
+main = hspec spec
+
+exampleAdvisory :: ByteString
+exampleAdvisory = [r|
```

Comment:
I was quite confused by `Text.RawString.QQ.r` and `QuasiQuotes` - is this the normal way to define multi-line strings?

Reply:
It can actually do much more than that. For example, there is a HTML templating engine using QuasiQuotes named Shakespeare (https://www.yesodweb.com/book/shakespearean-templates).